### PR TITLE
[docs] Add a changelog for 1.10.0

### PR DESCRIPTION
**Pull request by Farfurix**
### TODO
- [ ] Fix the release dates: the changelog and the website (https://github.com/DevExpress/testcafe/pull/5766)

### Changes
[v1.10.0-rc.8](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.8) (no changes)
[v1.10.0-rc.7](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.7) (no changes)
[v1.10.0-rc.6](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.6) (no changes)
[v1.10.0-rc.5](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.5) (no changes)
[v1.10.0-rc.4](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.4)
[v1.10.0-rc.3](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.3)
[v1.10.0-rc.2](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.2)
[v1.10.0-rc.1](https://github.com/DevExpress/testcafe/releases/tag/v1.10.0-rc.1)
[v1.9.5-rc.1](https://github.com/DevExpress/testcafe/releases/tag/v1.9.5-rc.1)
[v17.1.26](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.26) (no changes)
[v17.1.25](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.25)
[v17.1.24](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.24)
[v17.1.23](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.23)
[v17.1.22](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.22)
[v17.1.21](https://github.com/DevExpress/testcafe-hammerhead/releases/tag/v17.1.21)

### CODE REVIEW

**Review by VasilyStrelyaev — APPROVED** [review]

**Review by andnaz — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,125 @@
 # Changelog
 
+## v1.10.0 (2020-12-1)
+
+### Enhancements
+
+#### Window Resize and Screenshot Support for Child Windows in Chrome ([PR #5661](https://github.com/DevExpress/testcafe/pull/5661), [PR #5567](https://github.com/DevExpress/testcafe/pull/5567))
+
+You can now use the following actions in Google Chrome when the test context is switched to a [child window](https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/multiple-browser-windows.html):
```
Comment (dirk-pieterse) [inline CHANGELOG.md:9]:
```suggestion
You can now use the following actions in Google Chrome when you switch the test context to a [child window](https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/multiple-browser-windows.html):
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,125 @@
...
+
+* the [tsConfigPath](https://devexpress.github.io/testcafe/documentation/reference/configuration-file.html#tsconfigpath) configuration file property
+
+    ```json
+    {
+        "tsConfigPath": "path/to/config.json"
+    }
+    ```
+
+In v1.10.0, we introduced a new easy-to-use API that allows you to specify the compiler options in the command line, API or TestCafe configuration file, without creating a separate JSON file. The new API is also designed to accept options for more compilers (for instance, Babel) in the future releases.
```
Comment (dirk-pieterse) [inline CHANGELOG.md:41]:
```suggestion
In v1.10.0, we introduced a new easy-to-use API that allows you to specify the compiler options in the command line, API or TestCafe configuration file, without creating a separate JSON file. The new API is also designed to accept options for more compilers (for instance, Babel) in future releases.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,125 @@
...
+testcafe chrome my-tests --compiler-options typescript.configPath='path/to/config.json'
+```
+
+In v1.10.0, you can customize TypeScript compiler options only.
+
+For more information, see [TypeScript and CoffeeScript](https://devexpress.github.io/testcafe/documentation/guides/concepts/typescript-and-coffeescript.html).
+
+#### Added a Selector Method to Access Shadow DOM ([PR #5560](https://github.com/DevExpress/testcafe/pull/5560) by [@mostlyfabulous](https://github.com/mostlyfabulous))
+
+This release introduces the [selector.shadowRoot](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/shadowroot.html) method that allows you to easily access and interact with the shadow DOM elements. This method returns a shadow DOM root hosted in the selector's matched element.
```
Comment (dirk-pieterse) [inline CHANGELOG.md:85]: "that allows you to easily access and interact" I would advise you not to use 'easily' without explaining why it is easily.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,125 @@
...
+
+test('Get text within shadow tree', async t => {
+    const shadowRoot = Selector('div').withAttribute('id', 'shadow-host').shadowRoot();
+    const paragraph  = shadowRoot.child('p');
+
+    await t.expect(paragraph.textContent).eql('This paragraph is in the shadow tree');
+});
+```
+
+Note that you must chain other [selector methods](https://devexpress.github.io/testcafe/documentation/guides/basic-guides/select-page-elements.html#member-tables) to [selector.shadowRoot](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/shadowroot.html) in order to access elements in the shadow DOM. You cannot interact with the root element (an error occurs if you specify `selector.shadowRoot` as an action's target element).
```
Comment (dirk-pieterse) [inline CHANGELOG.md:101]:
```suggestion
Note that you should chain other [selector methods](https://devexpress.github.io/testcafe/documentation/guides/basic-guides/select-page-elements.html#member-tables) to [selector.shadowRoot](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/shadowroot.html) to access elements in the shadow DOM. You cannot interact with the root element (an error occurs if you specify `selector.shadowRoot` as an action's target element).
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,125 @@
...
+
+### Bug Fixes
+
+* Browsers now restart correctly on BrowserStack when connection is lost ([#5238](https://github.com/DevExpress/testcafe/issues/5238))
```
Comment (dirk-pieterse) [inline CHANGELOG.md:105]:
```suggestion
* Browsers now restart correctly on BrowserStack when the connection is lost ([#5238](https://github.com/DevExpress/testcafe/issues/5238))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,125 @@
...
+* Fixed an error that occurs if a child window is opened in an `iframe` ([#5033](https://github.com/DevExpress/testcafe/issues/5033))
+* TestCafe can now switch between the child and parent windows after the parent window is reloaded ([#5463](https://github.com/DevExpress/testcafe/issues/5463), [#5597](https://github.com/DevExpress/testcafe/issues/5597))
+* Fixed an issue when both touch and mouse events fired on mobile devices although the mouse event was prevented in page code ([#5380](https://github.com/DevExpress/testcafe/issues/5380))
```
Comment (dirk-pieterse) [inline CHANGELOG.md:108]:
```suggestion
* Fixed an issue when touch and mouse events fired on mobile devices even though the mouse event was prevented in page code ([#5380](https://github.com/DevExpress/testcafe/issues/5380))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,125 @@
...
+* Cross-domain `iframes` are now focused correctly in Safari ([#4793](https://github.com/DevExpress/testcafe/issues/4793))
+* Fixed an excessive warning displayed when an assertion is executed in a loop or against an element returned by a `selector.xxxSibling` method ([#5449](https://github.com/DevExpress/testcafe/issues/5449), [#5389](https://github.com/DevExpress/testcafe/issues/5389))
+* Cross-domain `iframe` source links now have the correct protocol when SSL is used ([PR testcafe-hammerhead/#2478](https://github.com/DevExpress/testcafe-hammerhead/pull/2478))
+* A page error is no longer emitted if the destination server responded with the `304` status ([#5025](https://github.com/DevExpress/testcafe/issues/5025))
+* Fixed an issue when TestCafe could not authenticate to websites that use MSAL ([#4834](https://github.com/DevExpress/testcafe/issues/4834))
```
Comment (dirk-pieterse) [inline CHANGELOG.md:113]:
```suggestion
* Fixed an issue when TestCafe could not authenticate websites that use MSAL ([#4834](https://github.com/DevExpress/testcafe/issues/4834))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,125 @@
...
+* The `srcdoc` attributes for `iframes` are now processed ([testcafe-hammerhead/#1237](https://github.com/DevExpress/testcafe-hammerhead/issues/1237))
+* The `authorization` header is now preserved in response headers of fetch requests ([testcafe-hammerhead/#2334](https://github.com/DevExpress/testcafe-hammerhead/issues/2334))
+* The `document.title` for an `iframe` without `src` can now be obtained correctly in Firefox ([PR testcafe-hammerhead/#2466](https://github.com/DevExpress/testcafe-hammerhead/pull/2466))
```
Comment (dirk-pieterse) [inline CHANGELOG.md:116]:
```suggestion
* The `document.title` for an `iframe` without `src` can now be correctly obtained in Firefox ([PR testcafe-hammerhead/#2466](https://github.com/DevExpress/testcafe-hammerhead/pull/2466))
```

File: docs/articles/documentation/guides/advanced-guides/multiple-browser-windows.md [hunk 9]
```diff
@@ -7,15 +7,15 @@ permalink: /documentation/guides/advanced-guides/multiple-browser-windows.html
 
 The TestCafe API includes methods that open, close, and switch between browser windows. You can test websites with pop-up windows and OAuth login forms, debug complex multi-window applications, or run multiple instances of the same web app side-by-side.
 
-⚠ This is a **beta** feature. Browser support is limited to local instances of Chrome and Firefox. You cannot resize child windows. You cannot take screenshots or videos of child windows. The available functionality is subject to further revisions. Please do not use this feature in production environments.
+⚠ This is a **beta** feature. Browser support is limited to local instances of Chrome and Firefox. You can resize child windows or capture them for screenshots in Chrome only. You cannot record videos of child windows. The available functionality is subject to further revisions. Please do not use this feature in production environments.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/guides/advanced-guides/multiple-browser-windows.md:10]:
```suggestion
⚠ This is a **beta** feature. Browser support is limited to local instances of Chrome and Firefox. You can resize child windows or make screenshots of them in Chrome only. You cannot record videos of child windows. The available functionality is subject to further revisions. Please do not use this feature in production environments.
```

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**